test/e2e/helpers.go

```diff
@@ -100,14 +98,45 @@ func CreateK8sObjectWithRetry[T kruntime.Object](
 // This function takes a delete function like clients.Kubernetes.CertificatesV1().CertificateSigningRequests().Delete
 // and the parameters for it. It then makes the call with some retry logic.
 //
-// By default the call is retried for 5s with a 250ms interval.
+// By default the call is retried for 5mins with a 250ms interval.
 func DeleteK8sObjectWithRetry(
 	ctx context.Context, delete K8sDeleteFunc, name string, options metav1.DeleteOptions,
-) {
+) (err error) {
 	Eventually(func(g Gomega, ctx context.Context) {
 		err := delete(ctx, name, options)
 		g.Expect(err).NotTo(HaveOccurred())
 	}).WithContext(ctx).WithTimeout(DefaultTimeout).WithPolling(PollingInterval).Should(Succeed())
+	return
+}
+
+type AllowedCleanUpAPIInterface[T kruntime.Object] interface {
+	Get(ctx context.Context, name string, opts metav1.GetOptions) (T, error)
+	Delete(ctx context.Context, name string, options metav1.DeleteOptions) error
+}
+
+// This function takes a client that knows how to issue a GET and DELETE call for a given resource.
+// It then issues a delete request then and polls the API to until the resource is no longer found.
+//
+// Note: If the DELETE request receives a 404 we assume the resource has been cleaned up successfully.
+//
+// By default the call is retried for 10mins with a 1s interval.
+func CleanupK8sResource[T kruntime.Object](
+	ctx context.Context, client AllowedCleanUpAPIInterface[T], name string,
+) {
```

Comment:
This might seem like overkill (an argument I'm open to despite what I'm about to say) but IMO this is a perfect example of what we can do with generics. The alternative to this function would be 5-6 near identical functions i.e. CleanUp{Pods/Services/Namespace...}.

Let me know what you think